crates/languages/src/rust.rs

```diff
@@ -542,12 +542,11 @@ fn retrieve_package_id_and_bin_name_from_metadata(
     metadata: CargoMetadata,
     abs_path: &Path,
 ) -> Option<(String, String)> {
-    let abs_path = abs_path.to_str()?;
-
     for package in metadata.packages {
         for target in package.targets {
             let is_bin = target.kind.iter().any(|kind| kind == "bin");
-            if target.src_path == abs_path && is_bin {
```

Comment:
On windows, the `target.src_path` could be `C:\\A\\B\\main.rs`, and `abs_path` could be `C:\\A\\B/main.rs`, so a simple string comparing maybe always return false.